host/tests/dsClasses/dsDisplay.py

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/env python3
+#** *****************************************************************************
+# *
+# * If not stated otherwise in this file or this component's LICENSE file the
+# * following copyright and licenses apply:
+# *
+# * Copyright 2024 RDK Management
+# *
+# * Licensed under the Apache License, Version 2.0 (the "License");
+# * you may not use this file except in compliance with the License.
+# * You may obtain a copy of the License at
+# *
+# *
+# http://www.apache.org/licenses/LICENSE-2.0
+# *
+# * Unless required by applicable law or agreed to in writing, software
+# * distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+# *
+#* ******************************************************************************
+
+import yaml
+import os
+import sys
+import re
+from enum import Enum, auto
+
+# Add parent outside of the class directory
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(dir_path+"/../")
+
+from raft.framework.plugins.ut_raft.configRead import ConfigRead
+from raft.framework.plugins.ut_raft.utSuiteNavigator import UTSuiteNavigatorClass
+from raft.framework.plugins.ut_raft.interactiveShell import InteractiveShell
+
+
+class dsVideoPortType(Enum):
+    dsVIDEOPORT_TYPE_RF     = 0
+    dsVIDEOPORT_TYPE_BB =auto()
+    dsVIDEOPORT_TYPE_SVIDEO =auto()
+    dsVIDEOPORT_TYPE_1394 =auto()
+    dsVIDEOPORT_TYPE_DVI =auto()
+    dsVIDEOPORT_TYPE_COMPONENT =auto()
+    dsVIDEOPORT_TYPE_HDMI =auto()
+    dsVIDEOPORT_TYPE_HDMI_INPUT =auto()
+    dsVIDEOPORT_TYPE_INTERNAL =auto()
+
+class dsDisplayEvent(Enum):
+    dsDISPLAY_EVENT_CONNECTED = 0
+    dsDISPLAY_EVENT_DISCONNECTED = auto()
+    dsDISPLAY_RXSENSE_ON = auto()
+    dsDISPLAY_RXSENSE_OFF = auto()
+    dsDISPLAY_HDCPPROTOCOL_CHANGE = auto()
+    dsDISPLAY_EVENT_MAX = auto()
+
+class dsVideoAspectRatio(Enum):
+    dsVIDEO_ASPECT_RATIO_4x3 = auto()
+    dsVIDEO_ASPECT_RATIO_16x9 = auto()
+    dsVIDEO_ASPECT_RATIO_MAX = auto()
+
+
+class dsDisplayClass():
+
+    """
+    Device Settings Display Class
+    This module provides common extensions for device Settings Display Module.
+    """
+    moduleName = "dsDisplay"
+    menuConfig = dir_path+ "/dsDisplay_test_suite.yml"
+    testSuite = "L3 dsDisplay"
+    validationConfig = dir_path + "/../L3_TestCases/dsDisplay/dsDisplay_test_validation.yml"
```

Comment:
validationConfig is test specific configuration file, not the dsDisplay.py class, move this parsing to dsDisplayHelperClass.py